### PR TITLE
docs(documentai): Add Initial Sample for Document AI

### PR DESCRIPTION
**Pull request by holtskinner**
<s> NOTE: The test project will need the following API enabled

- `documentai.googleapis.com`

And a Document OCR Processor will need to be created in the project

https://cloud.google.com/document-ai/docs/create-processor

And the Processor ID should replace the `aaaaaaaaa` in the `process_document_test.go` file </s>

EDIT: Not needed for quickstart only

### CODE REVIEW

File: documentai/snippets/quickstart.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+	projectID := "PROJECT_ID"
+	locationID := "us"
+	processorID := "aaaaaaaa" // Create a Processor before running sample
+	filePath := "invoice.pdf"
+	mimeType := "application/pdf"
+
+	ctx := context.Background()
+
+	endpoint := fmt.Sprintf("%s-documentai.googleapis.com:443", locationID)
```
Comment (telpirion) [inline documentai/snippets/quickstart.go:41]: Don't switch between `fmt` and `log` in the same sample. Pick one and stick with it.

**Review by telpirion — COMMENTED** [review]
This sample needs a test.

File: documentai/snippets/quickstart.go (hunk 1 above)
Comment (holtskinner) [reply documentai/snippets/quickstart.go:41]: log is only used for error reporting. All of the other samples I looked at followed this pattern

Comment (holtskinner) [reply documentai/snippets/quickstart.go:41]: nvm, found this in the contributing guide https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#return-errors

**Comment by telpirion** [comment]
I believe that the issue in merging here is caused by a dependency error else where. #2853 will hopefully fix.

**Comment by holtskinner** [comment]
Hmm, it seems like the build is still failing in the same spot even after the updates...

**Comment by holtskinner** [comment]
Created new PR #2856

**Pull request closed without merging**